File: generated-code/example/kotlin/src/main/kotlin/Runner.kt
import java.io.*

fun main(args: Array<String>) {
    if (args.size != 2) {
        throw RuntimeException("Pass input and output as parameters")
    }
    val inputFile = args[0]
    val outputFile = args[1]

    val inputStream: InputStream = BufferedInputStream(FileInputStream(inputFile))
    val input: model.Structure = model.Structure.readFrom(inputStream)

    val outputStream: OutputStream = BufferedOutputStream(FileOutputStream(outputFile))
    input.writeTo(outputStream)
    outputStream.flush()
    outputStream.close()
}

File: generated-code/codecraft/fsharp/Runner.fs
namespace TransGenTest

open System
open System.IO

module Runner =

    [<EntryPoint>]
    let main argv =
        if argv.Length <> 2 then
            failwith "Pass input and output as parameters"
        let inputFile = argv.[0]
        let outputFile = argv.[1]

        use inputStream = new FileStream(inputFile, FileMode.Open)
        use reader = new BinaryReader(inputStream)
        let input = Model.PlayerView.readFrom(reader)

        Console.WriteLine(input)

        use outputStream = new FileStream(outputFile, FileMode.Create)
        use writer = new BinaryWriter(outputStream)
        input.writeTo(writer)

        0